src/env/config_paths.rs

```diff
@@ -0,0 +1,174 @@
+use std::{
+    ffi::OsStr,
+    path::{Path, PathBuf},
+};
+
+use once_cell::sync::Lazy;
+
+use crate::{FLOG, FLOGF};
+
+use super::ConfigPaths;
+
+const DOC_DIR: &str = env!("DOCDIR");
+const DATA_DIR: &str = env!("DATADIR");
+const DATA_DIR_SUBDIR: &str = env!("DATADIR_SUBDIR");
+const SYSCONF_DIR: &str = env!("SYSCONFDIR");
+const BIN_DIR: &str = env!("BINDIR");
+
+pub static CONFIG_PATHS: Lazy<ConfigPaths> = Lazy::new(|| {
+    // Read the current executable and follow all symlinks to it.
+    determine_config_directory_paths(std::env::current_exe().unwrap().canonicalize().unwrap())
```

Comment:
This would crash fish on startup, unconditionally, on platforms that don't support std::env::current_exe().

Reply:
https://github.com/rust-lang/rust/issues/60560 - apparently it currently silently returns a wrong path, but should probably fail instead. OpenBSD downstream applies a patch to rust, so if you use the distro's rust compiler it will produce a crashing fish.

We have our own way to resolve the binary, and it doesn't work on OpenBSD - #9086. It's what `status fish-path` uses.

Reply:
There seems to be very little we can do for OpenBSD in general, other than checking existence of the path if it is returned and falling back to `argv[0]` on failure.

Reply:
Done in https://github.com/fish-shell/fish-shell/pull/11195/commits/fa02518c4e8b40a498c1d7bcc7403289129ab727